### PR TITLE
Refactor OpenTelemetry.Extensions.Hosting readme

### PR DESCRIPTION
**Pull request by alanwest**

### CODE REVIEW

File: src/OpenTelemetry.Extensions.Hosting/README.md [hunk 1]
```diff
@@ -111,6 +80,25 @@ var app = appBuilder.Build();
...
+  to automatically start tracing services in the supplied
+  [IServiceCollection](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.iservicecollection).
+
+* `AddOpenTelemetryMetrics`: Configure OpenTelemetry and register an
+  [IHostedService](https://learn.microsoft.com/dotnet/api/microsoft.extensions.hosting.ihostedservice)
+  to automatically start metric services in the supplied
+  [IServiceCollection](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.iservicecollection).
+
+These methods were marked obsolete and later removed. You should migrate your
+code to the new `AddOpenTelemetry` method documented above.
```
Comment (cijothomas) [inline src/OpenTelemetry.Extensions.Hosting/README.md:100]: i wonder if we should give an example startup code for this? or the PR which did this for our example one? or maybe link to old and new version of our example app?

**Review by cijothomas — APPROVED** [review]

File: src/OpenTelemetry.Extensions.Hosting/README.md [hunk 2]
```diff
@@ -21,8 +21,6 @@ and metrics (`MeterProvider`) in [ASP.NET
 
 ## Extension method reference
 
-### Current OpenTelemetry SDK v1.4.0 and newer extensions
```
Comment (cijothomas) [inline src/OpenTelemetry.Extensions.Hosting/README.md:24]: consider a link to the example app, which shows this in action.

Comment (CodeBlanch) [reply src/OpenTelemetry.Extensions.Hosting/README.md:24]: There is a usage section below just not shown on the diff unless you expand it

Comment (CodeBlanch) [reply src/OpenTelemetry.Extensions.Hosting/README.md:24]: Or if you are saying in addition to that a link would be good, find by me 😄

**Review by CodeBlanch — APPROVED** [review]

File: src/OpenTelemetry.Extensions.Hosting/README.md (hunk 2 above)
Comment (cijothomas) [reply src/OpenTelemetry.Extensions.Hosting/README.md:24]: yea on top of the basic usage shown here.

Comment (cijothomas) [reply src/OpenTelemetry.Extensions.Hosting/README.md:24]: the example, if followed correctly, will help avoid mistakes like adding ActivitySource to DI etc. So good to link from this

File: src/OpenTelemetry.Extensions.Hosting/README.md (hunk 1 above)
Comment (CodeBlanch) [reply src/OpenTelemetry.Extensions.Hosting/README.md:100]: We could link to https://github.com/open-telemetry/opentelemetry-dotnet/pull/4071 where there are some migration steps.

Comment (alanwest) [reply src/OpenTelemetry.Extensions.Hosting/README.md]: > We could link to #4071 where there are some migration steps.

oops didn't see this comment before I updated it. I just added links to the old (1.3.2 version) and new (main). What do you all think?

**Review by cijothomas — APPROVED** [review]

**Pull request merged**